packages/ses/src/compartment-shim.js

```diff
@@ -170,9 +200,19 @@ export class Compartment {
     } = options;
     const localTransforms = [...transforms];
 
-    const { globalTransforms, globalObject } = privateFields.get(this);
+    const {
+      globalTransforms,
+      globalObject,
+      globalLexicals,
+    } = privateFields.get(this);
     const realmRec = getCurrentRealmRec();
-    return performEval(realmRec, source, globalObject, endowments, {
+
+    // TODO just pass globalLexicals as globalObject
+    // https://github.com/Agoric/SES-shim/issues/365
+    const localObject = create(null, getOwnPropertyDescriptors(globalLexicals));
+    defineProperties(localObject, getOwnPropertyDescriptors(endowments));
```

Comment:
If there's a collision between the globalLexicals and the endowments, this will throw. Is this intended?

The shared globals plus the endowments will be on the globalObject. Why are you putting the endowments on the localObject as well?

It is confusing that you sample the globalLexicals using `...` semantics but sample endowments using `defineProperty/getOwnPropertyDescriptors` semantics (all properties, whether enumerable or not. Preserve accessors). Is this what you want?

These are per-evaluation endowments? Is this only for live bindings?

Given that these endowments are much more like a per-evaluation equivalent to per-compartment globalLexicals than they are like a per-evaluation equivalent of per-compartment endowments, can we call these something other than endowments?

Reply:
As discussed, `globalLexicals` should be copied to initialize the `localObject` using `...`

Reply:
A test revealed that this will not be suitable. `globalLexicals` are immutable but per-evaluate `endowments` are mutable. It won’t be sufficient to combine them and freeze the whole. We need the property descriptors from `globalLexicals` (which was frozen by the constructor) to distinguish them from the per-evaluate endowments, for which there are pre-existing tests that require the endowments properties to be carried wholesale.

This will become irrelevant after removing per-evaluate endowments and we can simplify in that change.